src/components/no-periods.cjsx

```diff
@@ -9,14 +10,34 @@ NoPeriods = React.createClass
   propTypes:
     courseId: React.PropTypes.string.isRequired
     noPanel:  React.PropTypes.bool
+    link:  React.PropTypes.bool
+
+  getDefaultProps: ->
+    link: true
+
+  getMessage: ->
+    [
+      'Please add at least one '
+      <CourseGroupingLabel lowercase courseId={@props.courseId} />
+      ' to the course.'
+    ]
 
```

Comment:
React complains because these don't have `keys`.  I think we'll have to make the bare strings into `<span key="stuff">`, and add a key to CourseGroupingLabel to make it happy, React will insert span tags anyways so it's really the same.